akka-persistence/src/main/scala/akka/persistence/AtLeastOnceDelivery.scala

```diff
@@ -380,6 +384,8 @@ abstract class AbstractPersistentActorWithAtLeastOnceDelivery extends AbstractPe
    * This method will throw [[AtLeastOnceDelivery.MaxUnconfirmedMessagesExceededException]]
    * if [[numberOfUnconfirmed]] is greater than or equal to [[#maxUnconfirmedMessages]].
    */
-  def deliver(destination: ActorPath, deliveryIdToMessage: akka.japi.Function[java.lang.Long, Object]): Unit =
+  def deliver(destination: ActorPath, deliveryIdToMessage: akka.japi.function.Function[java.lang.Long, Object]): Unit =
     super.deliver(destination, id ⇒ deliveryIdToMessage.apply(id))
+
+  // FIXME @deprecated deliver with akka.japi.Function. How would java8 know which one to pick when param is lambda?
```

Comment:
Bad news on this one - javac won't be able to figure out which target type you want this lambda to be expanded to when it sees `deliver(..., in -> { ... })` because both match (arity + types + return type).

The error would be along the lines of:

```
Example.java:25: error: reference to doIt is ambiguous
        whoops.doIt(() -> {
              ^
  both method doIt(Fun) in Whoops and method doIt(FunBetter) in Whoops match
```

(It's solvable as `deliver(..., (akka.japi.function.Function<Long, Object>)(l) -> l)` but that's not really helping...)